### PR TITLE
With Firebase Hosting Example

### PR DESCRIPTION
**Pull request by jthegedus**
I managed to solve many of the problems people were having with Next.js on Cloud Functions for Firebase. EG:
    - https://github.com/zeit/next.js/issues/2017
    - https://github.com/zeit/next.js/issues/820
    - https://github.com/zeit/next.js/issues/2467
    - https://github.com/firebase/firebase-functions/issues/72

I'd like to contribute it here.

I know that everyone maintaining this repo and the [Firebase Functions](https://github.com/firebase/firebase-functions) repo are busy and don't have the time to support a myriad of deployment methods or JS frameworks. Given that, I'm happy to tackle any Firebase Hosting issues if you wish to redirect people to [my example Firebase repo](https://github.com/jthegedus/firebase-functions-next-example). If and when we diagnose issues as being on the Next.js side I can bring the concerns to this repo.

I also have a thorough [Medium post](https://medium.com/@jthegedus/next-js-on-cloud-functions-for-firebase-with-firebase-hosting-7911465298f2) about this setup and the challenges overcome, but wasn't sure of the contributors guide for example repositories and linking externally. If it's okay to add a link to this example I think it will answer a lot of people's questions when they Google their way here.

Thanks for the cool framework! :smile:

### CODE REVIEW

**Comment by jthegedus** [comment]
This PR makes a few changes I wish to highlight:

- It adds npm5 lockfile to the gitignore.
- It renames the previous Firebase example from `with-firebase` to `with-firebase-authentication` as it only covers using Firebase Authentication and not Hosting.

File: examples/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 **/yarn.lock
+**/package-lock.json
```
Comment (timneutkens) [inline examples/.gitignore]: no need for this 😄 https://github.com/zeit/next.js/blob/v3-beta/.gitignore#L7

**Review by timneutkens — COMMENTED** [review]
Looks good, if you remove the package-lock.json stuff I'll merge it 👍 ❤️

**Review by timneutkens — APPROVED** [review]

File: examples/.gitignore (hunk 1 above)
Comment (jthegedus) [inline examples/.gitignore]: Didn't see that. Updated! Cheers!

**Pull request merged**